app/forms/change_case_visibility_form.rb

```
@@ -0,0 +1,24 @@
+class ChangeCaseVisibilityForm
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Serialization
+
+  attribute :case_type
+  attribute :old_visibility
+  attribute :new_visibility
+  attribute :rationale
+
+  validates_inclusion_of :new_visibility, in: %w[restricted unrestricted]
+  validate :new_visibility_is_different
+
+  def self.from(investigation)
+    old_visibility = investigation.is_private? ? "restricted" : "unrestricted"
+    new(case_type: investigation.case_type, old_visibility: old_visibility)
+  end
+
+private
+
+  def new_visibility_is_different
+    errors.add(:new_visibility, :same_as_existing, case_type: case_type, visibility: new_visibility) if new_visibility == old_visibility
```

I think this is a change in behaviour, let's check with @DH02 that we want to throw an error instead of accepting the form when the restriction hasn't changed.

---

Ignore this, I didn't look at the designs!

---

Cool, thanks @slorek, I think I will let Nas have a last look over this on Monday then merge then if he is happy!